src/main/java/uk/gov/hmcts/reform/feature/security/IdamUserAuthenticationFilter.java

```diff
@@ -0,0 +1,62 @@
+package uk.gov.hmcts.reform.feature.security;
+
+import org.springframework.http.HttpHeaders;
+import org.springframework.security.access.intercept.RunAsUserToken;
+import org.springframework.security.authentication.AnonymousAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.AuthenticationException;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.userdetails.User;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.web.authentication.AbstractAuthenticationProcessingFilter;
+import org.springframework.security.web.util.matcher.AntPathRequestMatcher;
+import uk.gov.hmcts.reform.feature.model.UserTokenDetails;
+
+import java.io.IOException;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import static org.springframework.http.HttpMethod.GET;
+
+public class IdamUserAuthenticationFilter extends AbstractAuthenticationProcessingFilter {
+
+    public IdamUserAuthenticationFilter(String pattern) {
+        super(new AntPathRequestMatcher(pattern, GET.name()));
+    }
+
+    @Override
+    public Authentication attemptAuthentication(
+        HttpServletRequest request,
+        HttpServletResponse response
+    ) throws AuthenticationException, IOException, ServletException {
+        String authorisationHeader = request.getHeader(HttpHeaders.AUTHORIZATION);
+
+        if (authorisationHeader != null) {
+            UserTokenDetails userTokenDetails = null;// todo parse jwt
```

Comment:
reading roles from JWT token shouldn't require much code (I think), so it may be worth adding it as part of this PR, as at the moment this has no tests and it's hard to say whether it's ok.

Reply:
wanted to leave any jwt action for separate pull request including unit testing. integration/functional will come in another pr once i will stick in the filter where it needs to be